src/rules/dynamic-import-chunkname.js

```diff
@@ -27,10 +35,19 @@ module.exports = {
   create(context) {
     const config = context.options[0];
     const { importFunctions = [] } = config || {};
-    const { webpackChunknameFormat = '[0-9a-zA-Z-_/.]+' } = config || {};
-
+    const { webpackChunknameFormat = '([0-9a-zA-Z-_/.]|\\[(request|index)\\])+' } = config || {};
+    const webpackMagicComments = [
+      ['webpackChunkName', `["']${webpackChunknameFormat}["']`],
+      ['webpackPrefetch', '(true|false|-?[0-9]+)'],
+      ['webpackPreload', '(true|false|-?[0-9]+)'],
+      ['webpackIgnore', '(true|false)'],
+      ['webpackInclude', '\\/.*\\/'],
+      ['webpackExclude', '\\/.*\\/'],
+      ['webpackMode', `["'](lazy|lazy-once|eager|weak)["']`],
+      ['webpackExports', `(['"]\\w+['"]|\\[(['"]\\w+['"], *)+(['"]\\w+['"]*)\\])`],
+    ];
     const paddedCommentRegex = /^ (\S[\s\S]+\S) $/;
-    const commentStyleRegex = /^( \w+: (["'][^"']*["']|\d+|false|true),?)+ $/;
+    const commentStyleRegex = generateRegexForMagicComments(webpackMagicComments);
```

Comment:
why are we doing regex magic here instead of just hardcoding each regex?

Reply:
> why are we doing regex magic here instead of just hardcoding each regex?

I though this way it's more clear what's happening and more readable overall. I feel like concatenating everything by hand in one huge regex is harder to grasp and more error prone (cause you need to do it by hand).

```
const commentStyleRegex = new RegExp(`^( ((webpackChunkName: ["']${webpackChunknameFormat}["'])|(webpackPrefetch: (true|false|-?[0-9]+))|(webpackPreload: (true|false|-?[0-9]+))|(webpackIgnore: (true|false))|(webpackInclude: \\/.*\\/)|(webpackExclude: \\/.*\\/)|(webpackMode: ["'](lazy|lazy-once|eager|weak)["'])|(webpackExports: (['"]\\w+['"]|\\[(['"]\\w+['"], *)+(['"]\\w+['"]*)\\]))),?)+ $`);
```

Reply:
While that’s true, dynamic regex construction is a huge code smell; i think it’s worth the clarity hit to avoid it.